pkg/monitor/monitor/monitor_manager.go

```diff
@@ -182,3 +209,13 @@ func (mm *MonitorManager) syncTidbMonitorRbac(monitor *v1alpha1.TidbMonitor) (*c
 
 	return sa, nil
 }
+
+// If the namespace in ClusterRef is empty, we would set the TidbMonitor's namespace in the default
+func setDefaultNamespaceForClusterRef(tm *v1alpha1.TidbMonitor) {
```

Comment:
Please do not mutate spec in controller, spec is user intent and by definition, is owned by user

Reply:
updated.